pkg/draft/context.go

```diff
@@ -65,3 +71,17 @@ func newAppContext(s *Server, req *rpc.UpRequest, out io.Writer) (*AppContext, e
 		vals: vals,
 	}, nil
 }
+
+func getulid() string { return <-ulidc }
+
+// A channel which returns build ulids.
+var ulidc = make(chan string)
+
+func init() {
+	rnd := rand.New(rand.NewSource(time.Now().UTC().UnixNano()))
+	go func() {
+		for {
+			ulidc <- ulid.MustNew(ulid.Timestamp(time.Now().UTC()), rnd).String()
```

Comment:
I'm constantly impressed with your code. This is actually much more performant than my suggestion from a very basic benchmark.

Current code as-is:

```
><> go test -bench=.
BenchmarkGetulid-8       3000000               443 ns/op
PASS
ok      github.com/Azure/draft/pkg/draft        1.983s
```

And my suggested changes actually make it about twice as slow.

```
><> go test -bench=.
BenchmarkGetulid-8        200000             10191 ns/op
PASS
ok      github.com/Azure/draft/pkg/draft        2.369s
```

I feel like I should be wearing a tinfoil hat for the rest of the day. Carry on!